test/test_cleanfigure.py

```diff
@@ -0,0 +1,1049 @@
+    plt.close("all")
+
+
+class Test_plottypes:
+    """Testing plot types found here https://matplotlib.org/3.1.1/tutorials/introductory/sample_plots.html"""
+
+    def test_plot(self):
+        """ """
+
+        x = np.linspace(1, 100, 20)
+        y = np.linspace(1, 100, 20)
+
+        with plt.rc_context(rc=RC_PARAMS):
+            fig, ax = plt.subplots(1, 1, figsize=(5, 5))
+            ax.plot(x, y)
+            ax.set_ylim([20, 80])
+            ax.set_xlim([20, 80])
+            raw = get_tikz_code()
+
+            cleanfigure.clean_figure(fig)
+            clean = get_tikz_code()
+
+            # Use number of lines to test if it worked.
+            # the baseline (raw) should have 20 points
+            # the clean version (clean) should have 2 points
+            # the difference in line numbers should therefore be 2
+            numLinesRaw = raw.count("\n")
+            numLinesClean = clean.count("\n")
+            assert numLinesRaw - numLinesClean == 18
+        plt.close("all")
+
+    def test_step(self):
+        """ """
+
+        x = np.linspace(1, 100, 20)
+        y = np.linspace(1, 100, 20)
+
+        with plt.rc_context(rc=RC_PARAMS):
+            fig, ax = plt.subplots(1, 1, figsize=(5, 5))
+            ax.step(x, y)
+            ax.set_ylim([20, 80])
+            ax.set_xlim([20, 80])
+            with pytest.warns(Warning):
+                cleanfigure.clean_figure(fig)
+        plt.close("all")
+
+    def test_scatter(self):
+        """ """
+        # TODO: scatter plots are represented through axes.collections. Currently, this is simply ignored and nothing is done.
+
+        x = np.linspace(1, 100, 20)
+        y = np.linspace(1, 100, 20)
+        with plt.rc_context(rc=RC_PARAMS):
+            fig, ax = plt.subplots(1, 1, figsize=(5, 5))
+            ax.scatter(x, y)
+            ax.set_ylim([20, 80])
+            ax.set_xlim([20, 80])
+            with pytest.warns(Warning):
+                cleanfigure.clean_figure(fig)
+        plt.close("all")
+
+    def test_bar(self):
+        """ """
+
+        x = np.linspace(1, 100, 20)
+        y = np.linspace(1, 100, 20)
+        with plt.rc_context(rc=RC_PARAMS):
+            fig, ax = plt.subplots(1, 1, figsize=(5, 5))
+            ax.bar(x, y)
+            ax.set_ylim([20, 80])
+            ax.set_xlim([20, 80])
+            with pytest.warns(Warning):
+                cleanfigure.clean_figure(fig)
+        plt.close("all")
+
+    def test_hist(self):
+        """creates same test case as bar"""
+
+        x = np.linspace(1, 100, 20)
+        y = np.linspace(1, 100, 20)
+        with plt.rc_context(rc=RC_PARAMS):
+            fig, ax = plt.subplots(1, 1, figsize=(5, 5))
+            ax.hist(x, y)
+            ax.set_ylim([20, 80])
+            ax.set_xlim([20, 80])
+            with pytest.warns(Warning):
+                cleanfigure.clean_figure(fig)
+        plt.close("all")
+
+    def test_plot3d(self):
+        """ """
```

Comment:
You can remove all those empty doc strings. (And perhaps also the octave-code ones, unless you use them yourself.)